### ISSUE TITLE
`proxyBasePath` requires leading slash to work correctly

### ISSUE THREAD

**Issue by thomasheartman** [Unleash/unleash-proxy#76]
### Describe the bug

When providing a `proxyBasePath`, it _must_ start with a leading slash to work correctly. The example in the docs does not have a leading slash.

Mentioned [in this Slack thread](https://unleash-community.slack.com/archives/C03GWTN7XMG/p1656578849919209). 

### Steps to reproduce the bug

1. Run the proxy as via Node.js
2. Specify `some/base` as the base path
3. try to curl http://localhost:3000/some/base/proxy
4. Get a 404

### Expected behavior

I'd expect the proxy to "do what I mean". Whether I use a leading or trailing slash in a path section shouldn't matter. These things always catches someone off guard, and in this case I really can't see any reason why it should matter. 

In other words, regardless of which of the following options you provide, the result should be the same (`/base/path/proxy`):
- `/base/path/`
- `/base/path`
- `base/path/`
- `base/path`

### SDK information (language and version)

0.10 of the Proxy

**Comment by thomasheartman** [comment]
You know, stale bot, I _do_ still want to do this; I just don't know when I'll have the time. But it _is_ a bug, so I'll pin it for now.
